### PR TITLE
Example of how a tile containing only a form should be structured - Improvement Day

### PR DESCRIPTION
**Pull request by rpokorny**
I found it ambiguous determining how a tile containing only a form should be structured.  After doing more research into how `header` and `footer` are supposed to work, I determined that the form and the tile cannot be the same element, because `form` is not a [Sectioning Content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#sectioning-content) element - that is, it does not create a semantic container for the `<header>` and `<footer>`, like `<section>` does.  This Gallery example proves that the styles work if you nest the `<form>` inside the` <section class="nx-tile">`, and asserts that that is the correct way to do it.

The other age-old question about this scenario – whether to use `nx-tile-footer` or `nx-form-footer` – is moot in this PR as it is based on my other branch where I merged them both into simply `nx-footer`.

### CODE REVIEW

File: gallery/src/styles/NxTile/NxTilesExamples.tsx [hunk 1]
```diff
@@ -7,11 +7,15 @@
...
+import NxTileFormExample from './NxTileFormExample';
+
 const NxSimpleTileCode = require('!!raw-loader!./NxSimpleTileExample.html').default,
     NxTileWithActionsCode = require('!!raw-loader!./NxTileWithActionsExample.html').default,
     NxTileWithSubtitleCode = require('!!raw-loader!./NxTileWithSubtitleExample.html').default,
     NxTileSubsectionCode = require('!!raw-loader!./NxTileSubsectionExample.html').default,
-    NxTileWithHorizontalRuleCode = require('!!raw-loader!./NxTileWithHorizontalRuleExample.html').default;
+    NxTileWithHorizontalRuleCode = require('!!raw-loader!./NxTileWithHorizontalRuleExample.html').default,
+    NxTileFormCode = require('!!raw-loader!./NxTileFormExample.tsx').default;
+
```
Comment (raveneyex) [inline gallery/src/styles/NxTile/NxTilesExamples.tsx:18]: These consecutive empty-lines are triggering the linter.

**Comment by raveneyex** [comment]
Shouldn't we have another example where a Form shares a tile with some other content? after all that's a scenario we've already seen.

**Comment by rpokorny** [comment]
> after all that's a scenario we've already seen.

Where have we already seen that?

File: gallery/src/styles/NxTile/NxTilesExamples.tsx (hunk 1 above)
Comment (rpokorny) [reply gallery/src/styles/NxTile/NxTilesExamples.tsx:18]: > These consecutive empty-lines are triggering the linter.

321020a3

**Comment by raveneyex** [comment]
> > after all that's a scenario we've already seen.
> 
> Where have we already seen that?

The new add waiver page is all within a single tile and has things that are not inputs.

**Comment by rpokorny** [comment]
> The new add waiver page is all within a single tile and has things that are not inputs.

It has things that are not inputs but if IIRC, the entire tile is semantically considered a form.  Recall this exchange: https://sonatype.slack.com/archives/CCAPS4KL5/p1596042575286300?thread_ts=1596041650.285200&cid=CCAPS4KL5.

I've just gone back and checked the DOM, and the entire tile contents is a form: 
image

**Comment by raveneyex** [comment]
> > The new add waiver page is all within a single tile and has things that are not inputs.
> 
> It has things that are not inputs but if IIRC, the entire tile is semantically considered a form. Recall this exchange: https://sonatype.slack.com/archives/CCAPS4KL5/p1596042575286300?thread_ts=1596041650.285200&cid=CCAPS4KL5.
> 
> I've just gone back and checked the DOM, and the entire tile contents is a form:
> image

Oh, ok.
I was under the impression we did that as a workaround for not having defined the whole form layout.

**Review by raveneyex — APPROVED** [review]

**Review by jlopez-sonatype — APPROVED** [review]

**Review by glenhunter — APPROVED** [review]
Interesting about a `<form>` not being a sectioning block. I did not know that.

**Pull request merged**